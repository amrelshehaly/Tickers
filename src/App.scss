
.container {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(200px, 2fr));
    gap: 15px;
    padding: 0.5em;
  }
  
  .item {
    /* Optional styling for grid items */
    padding: 20px;
    justify-content: center;
    display: flex;
    align-items: center;
  }

  /* Media query for smaller screens */
@media (max-width: 600px) {
    .container {
        grid-template-columns: repeat(auto-fit, minmax(150px, 2fr));
    }
  }