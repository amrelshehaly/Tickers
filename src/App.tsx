import TickerCard from "./common/tickerCard";
import SearchBar from "./common/SearchBar";
import "./App.scss";
import InfiniteScroll from "react-infinite-scroll-component";
import Alert from "./common/Alert";
import useTicker from "./hooks/useTicker";


function App() {
  const { fetchMore, response, loading, reFetchData, error } = useTicker()
  return ( 
    <>
      <SearchBar
        onChange={(e) => {
          console.log(e);
        }}
      />
      {error && <Alert msg={error?.error} refetch={reFetchData} />}
      {response.results && response.results.length > 0 && (
        <InfiniteScroll
          dataLength={response.results.length}
          next={() => fetchMore()}
          hasMore={true}
          loader={loading && <>loading ...</>}
          scrollThreshold={0.8}
        >
          <div className="container">
            {response.results.map((val, key) => (
              <div className="item" key={key}>
                <TickerCard companyName={val.name} name={val.ticker} />
              </div>
            ))}
          </div>
        </InfiniteScroll>
      )}
    </>
  );
}

export default App;
