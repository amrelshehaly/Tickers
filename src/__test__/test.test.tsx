import { describe, it } from 'vitest'
import { render } from '@testing-library/react';
import App from '../App'
import { http, HttpResponse  } from 'msw';
import { setupServer } from 'msw/node';
import result from './output.json'


const server = setupServer(
    http.get(`https://api.polygon.io/v3/reference/tickers?active=true&limit=5&apiKey=${import.meta.env.VITE_APIKEY}`, () => {
      return HttpResponse.json(result, { status: 200 });
    })
)
  

beforeAll(() =>  {
    server.listen();
})
afterAll(() => server.close())


describe('Testing the tickers and search functionality', () =>{

    it('Should render input searchbar', () => {
        const { container } = render(<App />)
        container.querySelector('.SearchWrapper')
        expect(container.firstChild).toMatchSnapshot()
    })

    it('should load the tickers', async ({ expect }) => {
        const { container } = render(<App />)
        container.querySelector('.item')
        expect(container.firstChild).toMatchSnapshot()
    })


})