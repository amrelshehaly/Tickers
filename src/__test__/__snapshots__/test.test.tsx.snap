// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the tickers and search functionality > Should render input searchbar 1`] = `
<div
  class="SearchWrapper"
>
  <input
    placeholder="Search for a stock"
    type="search"
  />
</div>
`;

exports[`Testing the tickers and search functionality > Should render tye searchbar 1`] = `
<div
  class="SearchWrapper"
>
  <input
    placeholder="Search for a stock"
    type="search"
  />
</div>
`;

exports[`Testing the tickers and search functionality > should load the tickers 1`] = `
<div
  class="SearchWrapper"
>
  <input
    placeholder="Search for a stock"
    type="search"
  />
</div>
`;
